components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryRecoveryDataStore.java

```diff
@@ -133,12 +128,64 @@ public UserRecoveryDataDO load(String code) throws IdentityException {
-        //To change body of implemented methods use File | Settings | File Templates.
+        Registry registry = null;
+        try {
+            registry = IdentityMgtServiceComponent.getRegistryService().
+                    getConfigSystemRegistry(PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
+            registry.beginTransaction();
+            String dataPath = IdentityMgtConstants.IDENTITY_MANAGEMENT_DATA;
+            Collection dataItems = (Collection)registry.get(dataPath);
+            for (int i = 0; i < dataItems.getChildren().length; i++) {
+                Resource currentResource = registry.get(dataItems.getChildren()[i]);
+                if (currentResource instanceof Collection) {
+                    String[] currentResourceChildren = ((Collection) currentResource).getChildren();
+                    for (int j = 0; j < currentResourceChildren.length; j++) {
+                        Resource innerResource = registry.get(currentResourceChildren[j]);
+                        if (innerResource.getProperty(SECRET_KEY).equals(recoveryDataDO.getSecret())) {
+                            registry.delete(currentResourceChildren[j]);
+                            return;
+                        }
+                    }
+                } else {
+                    if (currentResource.getProperty(SECRET_KEY).equals(recoveryDataDO.getSecret())) {
+                        registry.delete(dataItems.getChildren()[i]);
+                        return;
+                    }
+                }
+            }
+        } catch (RegistryException e) {
+            throw new IdentityException("Error while deleting resource after loading", e);
+        } finally {
+            if (registry != null) {
+                try {
+                    registry.commitTransaction();
+                } catch (RegistryException e) {
+                    log.error("Error while deleting resource after loading.", e);
+                }
+            }
+        }
+
     }
 
     @Override
     public void invalidate(String userId, int tenantId) throws IdentityException {
-        //To change body of implemented methods use File | Settings | File Templates.
+        Registry registry = null;
+        try {
+            registry = IdentityMgtServiceComponent.getRegistryService().
+                    getConfigSystemRegistry(PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
+            registry.beginTransaction();
+            deleteOldResourcesIfFound(registry, userId, IdentityMgtConstants.IDENTITY_MANAGEMENT_DATA);
+        } catch (RegistryException e) {
+            throw new IdentityException("Error while deleting the old confirmation code.", e);
+        } finally {
+            if (registry != null) {
+                try {
+                    registry.commitTransaction();
+                } catch (RegistryException e) {
+                    log.error("Error while deleting the old confirmation code \n" + e);
```

Comment:
Thanks for pointing, Fixed with https://github.com/wso2/carbon-identity/pull/1675